File: src/console/command/impl/ExitCommand.php
<?php

namespace pocketcloud\cloud\console\command\impl;

use pocketcloud\cloud\console\command\Command;
use pocketcloud\cloud\console\command\flag\CommandFlag;
use pocketcloud\cloud\console\command\sender\ICommandSender;
use pocketcloud\cloud\console\command\SubCommand;
use pocketcloud\cloud\PocketCloud;

final class ExitCommand extends Command {

    public function __construct() {
        parent::__construct("exit", "Stop the cloud");
        $this->addFlag(CommandFlag::short("y"));
        $this->addFlag(CommandFlag::long("yes"));
    }

    public function run(ICommandSender $sender, string $label, array $args, ?SubCommand $subCommand, array $flags): bool {
        if (($flags["y"] ?? false) || ($flags["yes"] ?? false)) PocketCloud::getInstance()->shutdown();
        else $this->waitForConfirmation($sender, "§bAre you sure you want to stop the cloud?", ["yes", "true", "y", "t"])->then(function (bool $response): void {
            if ($response) PocketCloud::getInstance()->shutdown();
        });

        return true;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/impl/ExitCommand.java
package de.pocketcloud.cloud.console.command.impl;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.command.Command;
import de.pocketcloud.cloud.console.command.ctx.CommandContext;
import de.pocketcloud.cloud.console.command.desc.CommandDescription;
import de.pocketcloud.cloud.console.command.flag.CommandFlag;
import de.pocketcloud.cloud.console.command.sender.CommandSender;

@CommandDescription(name = "exit", description = "Shuts down the cloud")
public final class ExitCommand extends Command {

    @Override
    public void prepare() {
        addFlag(CommandFlag.shortFlag("y", true));
    }

    @Override
    public boolean run(CommandSender sender, CommandContext ctx) {
        if (ctx.hasFlag("y")) {
            PocketCloud.instance().shutdown();
        } else {
            awaitConfirmation(sender, "Are you sure you want to §cshutdown §rthe cloud?")
                    .thenSuccess(r -> {
                        if (r) {
                            PocketCloud.instance().shutdown();
                        }
                    });
        }
        return true;
    }
}